lib/ImageHelper.php

```diff
@@ -216,6 +202,24 @@ static function add_filters() {
 	}
 
 	//-- end of public methods --//
+
+	
+	/**
+	 * Checks if attachment is an image before deleting generated files
+	 *
+	 * @param  int  $post_id   an attachment post id
+	 *
+	 */
+	static function delete_generated_if_image( $post_id ) {
+		if ( wp_attachment_is_image( $post_id ) ) {
+			$attachment = new Image($post_id);
+			if ( $attachment->file_loc ) {
+				ImageHelper::delete_generated_files($attachment->file_loc);
+			}
+		}
```

Comment:
This function now is just a wrapper for `delete_generated_files` method with additional checks. Maybe `delete_generated_if_image` could be incorporated into `delete_generated_files`, so we do not need two function. Maybe like this:

```php
/**
 * Deletes the auto-generated files for resize and letterboxing created by Timber.

 * @param string|int  $local_file   File URL or Attachment ID
 * 									ex: /var/www/wp-content/uploads/2015/my-pic.jpg
 *	                            	or: http://example.org/wp-content/uploads/2015/my-pic.jpg
 */
public static function delete_generated_files( $local_file ) {
	// Check if file is an attachment
	if (wp_attachment_is_image($local_file)) {
		$attachment = new Image($local_file);
		$local_file = $attachment->file_loc;
	}

	if (URLHelper::is_absolute($local_file)) {
		$local_file = URLHelper::url_to_file_system($local_file);
	}
```

And because this function misses a visibility statement as well, we could also add `public` to it.

What do you think?